### DISCUSSION TITLE
Parsing the 3D model

### DISCUSSION THREAD

**Discussion by ssh4net** [houmain/gpupad#11] [Q&A] [answered]
Houman, hello!

What the common workflow to parse arbitrary 3D model with uv and vertex normals into glsl shader in GPUpad project?
In samples only gltf one but it mostly js related example.

Thank you.

**Comment by houmain** [comment]
Hi Vlad, thank you for your long time interest in this project!
There is unfortunately no simple way yet. I intended to include a "custom action" which allows one to import glTF models into the session. It was never finished and does not seem to work at the moment.
There have not been new releases in a while, since I am working on a bigger update (a Vulkan renderer).

**Reply by ssh4net** [reply, marked as answer]
No problem. Already got how make this works. And made a small tool to convert OBJ files to a separate (per attribute) files:
https://github.com/ssh4net/OBJ2BIN

Later found that everything (except indexes) can be in the same file, but for prototyping separate files is fine 😀

**Reply by houmain** [reply]
Awesome! This is the first GPUpad related project (as far as I know).

**Reply by ssh4net** [reply]
I use GPUpad for prototyping different image processing shaders that I can use in my RawGL command line app. Quite specific use case, but I still think this much easier than using some game engines, or 3D apps...
This is a looong live stream I made with introduction into GPUpad, goal shaders and some image processing tricks https://www.youtube.com/live/iqFKEtDN1jU?si=5RaRDrjQ1qFSAaGa

**Reply by ssh4net** [reply]
@houmain are there any chance to add miniply and rapidobj to load PLY and OBJ files directly? This is a small and fastest c++ libraries.

And if possible, any chance to add some way to load meshes per frame (not a realtime).

**Reply by houmain** [reply]
I will see what I can do.
